src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -140,19 +140,84 @@ public async Task<SubscriptionState> AssertSubscriptionConnectionDetails(long id
+                
                 var whoseTaskIsIt = _db.WhoseTaskIsIt(topology, subscription, subscription);
+                
                 if (whoseTaskIsIt != _serverStore.NodeTag)
                 {
+                    var databaseTopologyAvailabilityExplanation = new Dictionary<string, string>();
+
+                    string generalState = string.Empty;
+                    if (_serverStore.Engine.CurrentState == RachisState.Candidate || _serverStore.Engine.CurrentState == RachisState.Passive)
+                    {
+                        generalState = $"Current node ({_serverStore.NodeTag}) is in {_serverStore.Engine.CurrentState.ToString()} state therefore, we can't answer who's task is it and returning null";
+                    }
+                    else
+                    {
+                        generalState = _serverStore.Engine.CurrentState.ToString();
+                    }
+                    databaseTopologyAvailabilityExplanation["NodeState"] = generalState;                    
+
+                    FillNodesAvailabilityReportForState(subscription, topology, databaseTopologyAvailabilityExplanation, stateGroup:topology.Rehabs, stateName:"rehab");
+                    FillNodesAvailabilityReportForState(subscription, topology, databaseTopologyAvailabilityExplanation, stateGroup: topology.Promotables, stateName: "promotable");
+
+                    //whoseTaskIsIt!= null && whoseTaskIsIt == subscription.MentorNode 
+                    foreach (var member in topology.Members)
+                    {
+                        if (whoseTaskIsIt != null)
+                        {
+                            if (whoseTaskIsIt == subscription.MentorNode && member == subscription.MentorNode)
+                            {
+                                databaseTopologyAvailabilityExplanation[member] = "Is the mentor node and a valid member of the topology, it should be the mento node";
+                            }
+                            else if (whoseTaskIsIt != null && whoseTaskIsIt != member)
+                            {
+                                databaseTopologyAvailabilityExplanation[member] = "Is a valid member of the topology, but not chosen to be the node running the subscription";
+                            }
+                            else if (whoseTaskIsIt == member)
+                            {
+                                databaseTopologyAvailabilityExplanation[member] = "Is a valid member of the topology and is chosen to be running the subscription";
+                            }
+                        }                        
+                        else
+                        {
+                            databaseTopologyAvailabilityExplanation[member] = "Is a valid member of the topology but was not chosen to run the subscription, we didn't find any other match either";
```

Comment:
Add here if we have a mentor node?

Reply:
When can this happen?

Reply:
When the node becomes passive in the middle of the subscription connection